pandas/tests/groupby/test_groupby.py

```diff
@@ -120,6 +126,52 @@ def checkit(dtype):
         for dtype in ['int64', 'int32', 'float64', 'float32']:
             checkit(dtype)
 
+    def test_basic_with_nan(self):
+
+        # GH 3729
+
+        def checkit(dtype):
+            data = Series(np.arange(9) // 3, index=np.arange(9), dtype=dtype)
```

Comment:
don't specify an index unless it is used explicity (e.g. here its just a range which is the default)